doc/iterations/2024.1/shapes/add_support_for_libraries.adoc

```diff
@@ -0,0 +1,39 @@
+= Add the support for libraries
+
+== Context
+
+Users want to have access to read only data (a.k.a libraries).
+
+== Key Result
+
+In the Details View, users could reference these read only data from their models' elements.
```

Comment:
You are detailing a use case where I can reference libraries with the details view but that's not even the first use case. I should be able to click on a template and have in my project a et of documents with objects referencing objects from libraries. As a result, without even a single interaction, I should be able to have a valid resource set.

The key technical result will be what's given to `EditingContextSearchService`, it does not involve in any way the Details view which is not involved at all in this feature.

Reply:
Another important key result is the fact that the libraries should be immutable, the read only aspect is important.

Reply:
done